internal/crypto/src/webcrypto/validators/ecdsa_validator.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright 2024 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use ecdsa::{signature::Verifier as EcdsaVerifier, Signature as EcdsaSignature};
+use p256::ecdsa::VerifyingKey as P256VerifyingKey;
+use p384::ecdsa::VerifyingKey as P384VerifyingKey;
+use spki::DecodePublicKey;
+
+use crate::raw_signature::{RawSignatureValidationError, RawSignatureValidator};
+
+/// An `EcdsaValidator` can validate raw signatures with one of the ECDSA
+/// signature algorithms.
+pub enum EcdsaValidator {
+    /// ECDSA with SHA-256
+    Es256,
+
+    /// ECDSA with SHA-384
+    Es384,
+    // ECDSA with SHA-512
+    // Es512, // not yet implemented (check with Colin)
```

Comment:
@cdmurph32 why was Es512 not implemented?

Reply:
Looks like the switch to Rustls here is a regression.  We currently support ES512 with SubtleCrypto.

Reply:
OK, that's the reason to keep the async path open. SubtleCrypto requires async.

Reply:
This code path is synchronous-only.